### PR TITLE
Fix bsc#1084275 to create a better example (for SLE12SP3)

### PR DESCRIPTION
**Pull request by tomschr**
Use Apache2 Web server, completely rewrite section

### CODE REVIEW

File: xml/docker_dockerizing_application.xml [hunk 1]
```diff
@@ -225,62 +225,123 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+            <calloutlist>
+             <callout arearefs="co.dockerfile.apache2.from">
+              <para>Base image, taken from <xref linkend="Building_Pre-build_Images"/>.</para>
+             </callout>
+             <callout arearefs="co.dockerfile.apache2.maintainer">
+              <para>Optional maintainer of the image.</para>
+             </callout>
+             <callout arearefs="co.dockerfile.apache2.copy">
+              <para>The copy instruction to copy your own data to the
+               server's directory. Currently, this lien is disabled due to
```
Comment (lightlace) [inline xml/docker_dockerizing_application.xml]: change 'lien' to 'line'

File: xml/docker_dockerizing_application.xml [hunk 2]
```diff
@@ -225,62 +225,123 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+               the hash mark in the first colomn.
```
Comment (lightlace) [inline xml/docker_dockerizing_application.xml]: replace 'colomn' with 'column'

File: xml/docker_dockerizing_application.xml [hunk 3]
```diff
@@ -225,62 +225,123 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+          </step>
+          <step>
+           <para>Run the image in <quote>detached</quote> mode:</para>
+           <screen>docker run --detach --interactive --tty &exampleuser_plain;/apache2</screen>
+           <para>Docker responds with the container ID, for example:</para>
+           <screen>e43fff4ae9832ecdb7677c058a73039d7610c32145a1d9b6ad0a4ed52b5c4dc7</screen>
+          </step>
+         </procedure>
+         <para>You can now access your data in <uri>http://localhost:80</uri>.
+         If you do not want to copy your data into your Docker container,
```
Comment (lightlace) [inline xml/docker_dockerizing_application.xml]: replace 'your' with 'the'. i think anything with 'you' is used too often.

Comment (sknorr) [reply xml/docker_dockerizing_application.xml]: @tomschr Please use \&docker;

File: xml/docker_dockerizing_application.xml [hunk 4]
```diff
@@ -225,62 +225,123 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+           <para>Open a Web browser and enter in the text field <uri>http://localhost:80/test.html</uri>.
+           You should see the output <literal>The Web Server is running</literal>.
+           </para>
+          </step>
+         </procedure>
+         <para>With the last procedure, you have built an image which was used
+         to test the build process and the Apache2 Web server. If everything
+         was successful, you can store your own data. Proceed as follows:</para>
+         <procedure>
+          <title>Creating a Docker Container with your Own Data</title>
```
Comment (sknorr) [inline xml/docker_dockerizing_application.xml]: Use \&docker;

File: xml/docker_dockerizing_application.xml [hunk 5]
```diff
@@ -225,62 +225,123 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+            server as it is not installed by default in the &slsa; Docker image.</para>
+          </step>
+          <step>
+           <para>Run the image in <quote>detached</quote> mode:</para>
+           <screen>docker run --detach --interactive --tty &exampleuser_plain;/apache2</screen>
+           <para>Docker responds with the container ID, for example:</para>
+           <screen>7bd674eb196d330d50f8a3cfc2bc61a243a4a535390767250b11a7886134ab93</screen>
+          </step>
+          <step>
+           <para>Open a Web browser and enter in the text field <uri>http://localhost:80/test.html</uri>.
```
Comment (sknorr) [inline xml/docker_dockerizing_application.xml]: text field -> location bar

File: xml/docker_dockerizing_application.xml [hunk 6]
```diff
@@ -225,62 +225,123 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+             described in <xref linkend="Building_Pre-build_Images"/>.</para>
+            </step>
+            <step>
+             <para>Save the <filename>Dockerfile</filename> from <xref
+              linkend="ex.docker.dockerfile.apache2"/> to a directory
+            <filename class="directory">docker</filename>.</para>
+            </step>
+            <step>
+             <para>Create your HTML files inside <filename>docker/data</filename>.
+             Anything you put in this directory is copied to the Docker image
```
Comment (sknorr) [inline xml/docker_dockerizing_application.xml]: Use \&docker;

File: xml/docker_dockerizing_application.xml [hunk 7]
```diff
@@ -225,62 +225,132 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+            option (here <literal>&exampleuser_plain;/apache2</literal>, but
+           you can use any name you want):</para>
+           <screen>sudo docker build -t &exampleuser_plain;/apache2 .</screen>
+           <para>Docker refreshes repositories and installs the Apache2
+            server as it is not installed by default in the &slsa; Docker image.</para>
+          </step>
+          <step>
+           <para>Run the image in <quote>detached</quote> mode:</para>
+           <screen>docker run --detach --interactive --tty &exampleuser_plain;/apache2</screen>
+           <para>Docker responds with the container ID, for example:</para>
```
Comment (sknorr) [inline xml/docker_dockerizing_application.xml]: Use \&deng; instead of "Docker"

File: xml/docker_dockerizing_application.xml [hunk 8]
```diff
@@ -225,62 +225,132 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+            </step>
+           </substeps>
+          </step>
+          <step xml:id="st.docker.apache2.build">
+           <para>Build the container. Set a tag for your image with the <option>-t</option>
+            option (here <literal>&exampleuser_plain;/apache2</literal>, but
+           you can use any name you want):</para>
+           <screen>sudo docker build -t &exampleuser_plain;/apache2 .</screen>
+           <para>Docker refreshes repositories and installs the Apache2
+            server as it is not installed by default in the &slsa; Docker image.</para>
```
Comment (sknorr) [inline xml/docker_dockerizing_application.xml]: Use \&docker;

File: xml/docker_dockerizing_application.xml [hunk 9]
```diff
@@ -225,62 +225,132 @@ ENTRYPOINT ["apache2ctl"]</screen>
...
+             </para>
+            </step>
+           </substeps>
+          </step>
+          <step xml:id="st.docker.apache2.build">
+           <para>Build the container. Set a tag for your image with the <option>-t</option>
+            option (here <literal>&exampleuser_plain;/apache2</literal>, but
+           you can use any name you want):</para>
+           <screen>sudo docker build -t &exampleuser_plain;/apache2 .</screen>
+           <para>Docker refreshes repositories and installs the Apache2
```
Comment (sknorr) [inline xml/docker_dockerizing_application.xml]: Use \&deng; instead of "Docker"

**Comment by tomschr** [comment]
Thanks! 

Here is the result as PDF: [docker.dockerizing.application_color_en.pdf](https://github.com/SUSE/doc-sle/files/2310514/docker.dockerizing.application_color_en.pdf)

**Comment by tomschr** [comment]
Thanks Stefan for the hints, it's fixed now in commit 49f6b0f.

As discussed with Tanja, due to an unresolved issue in bsc#1084275 for SLE15, we focus on SLE12SP3 for the time being. As such, we can't cherry-pick from develop.

**Pull request merged**